analyzers/tests/SonarAnalyzer.UnitTest/Extensions/AttributeDataExtensionsTest.cs

```diff
@@ -62,6 +63,95 @@ public class AttributeDataExtensionsTest
+        [DataRow("SomeString", typeof(int), false, 0)]
+        [DataRow("SomeNumberString", typeof(int), true, 42)]
+        public void TryGetAttributeValue_Arguments(string valueName, Type valueType, bool expectedSucess, object expectedResult)
+        {
+            var arguments = new Dictionary<string, object>
+            {
+                { "SomeBool", true },
+                { "SomeInt", 1_234_567 },
+                { "SomeByte", (byte)24 },
+                { "SomeString", "Text" },
+                { "SomeNumberString", "42" },
+                { "SomeNull", null },
+            };
+            TryGetAttributeValue_Arguments(valueName, valueType, expectedSucess, expectedResult, namedArguments: arguments);       // [Attr(SomeBool = true, SomeInt = 1_234_567, ..)]
+            TryGetAttributeValue_Arguments(valueName, valueType, expectedSucess, expectedResult, constructorArguments: arguments); // [Attr(SomeBool: true, SomeInt: 1_234_567, ..)]
+
+            static void TryGetAttributeValue_Arguments(string valueName,
+                                                       Type valueType,
+                                                       bool expectedSucess,
+                                                       object expectedResult,
+                                                       IDictionary<string, object> namedArguments = null,
+                                                       IDictionary<string, object> constructorArguments = null)
+            {
+                var attributeData = AttributeDataWithArguments(namedArguments, constructorArguments);
+                var tryGetAttributeValue = typeof(AttributeDataExtensions).GetMethod(nameof(AttributeDataExtensions.TryGetAttributeValue)).MakeGenericMethod(valueType);
+                var arguments = new object[] { attributeData, valueName, null };
+                var actualSuccess = tryGetAttributeValue.Invoke(null, arguments); // actualSuccess = attributeData.TryGetAttributeValue<valueType>(valueName, out valueType actualResult)
+                var actualResult = arguments[2]; // the out parameter value
+                actualSuccess.Should().Be(expectedSucess);
+                if (expectedResult == null)
+                {
+                    actualResult.Should().BeNull();
+                }
+                else
+                {
+                    actualResult.Should().BeOfType(expectedResult.GetType()).And.BeOfType(valueType);
+                    actualResult.Should().Be(expectedResult);
+                }
+            }
+        }
+
+        [TestMethod]
+        public void TryGetAttributeValue_ConstructorArgumentAndNamedArgumentNamedTheSame()
+        {
+            var attributeData = AttributeDataWithArguments(namedArguments: new Dictionary<string, object>
+            {
+                { "Result", true },
+            }, constructorArguments: new Dictionary<string, object>
+            {
+                { "Result", false },
+            });
```

Comment:
Changing the parameters from `IDictionary` to `Dictionary` allows this to be a single line
```suggestion
var attributeData = AttributeDataWithArguments(namedArguments: new() { { "Result", true } }, constructorArguments: new() { { "Result", false } });
```
In this rare case, I find the argument names useful

Reply:
The `new()` can be shortened on other places below too